zimui/src/components/ToolBar.vue

```diff
@@ -0,0 +1,108 @@
+<script setup lang="ts">
+import TopicSectionType from '@/types/TopicSection'
+import { PropType, onMounted, ref, watch } from 'vue'
+import TagButton from '../components/TagButton.vue'
+
+const props = defineProps({
+  topics: {
+    type: Object as PropType<TopicSectionType[]>,
+    required: true,
+  },
+})
+
+const tags = ref<
+  {
+    title: string
+    slug: string
+  }[]
+>()
+
+const showMore = ref(false)
+
+/**
+ * Limit number of tags based on max number of characters toolbar container can accomodate
+ * maxCharacters = toolbar container width / 7 (dividing factor should be adjusted based on font-size)
+ * @returns number of tags to display
```

Comment:
Mention here as well where the 150 and 25 magic values come from, will help next maintainer to understand the logic.

Reply:
Added the logic as comments